README.md

```diff
@@ -1,84 +1,105 @@
-# MyPackage
+# Toit Provision
 
-A template repository for creating a Toit package.
+Provision ESP32 connect to designated Wi-Fi access point by PC or App of mobile phone.
 
-## Toit package
-Use `toit.pkg describe` or `toit pkg describe` (depending on which Toit
-variant you use) to see how https://pkg.toit.io will extract package
-information from your repo when you publish the package.
+## 1. Tool and App
 
-Either add a `name: ...` entry to the package.yaml or change the title
-(first line) of this README to the package name.
+There is an issue in BLE provision of esp-idf, it is that esp-idf only supports blocking mode which cause BLE protocol stack blocks for about 4 seconds without response for any outside request. To skip this issue, you should modify provision source code of PC and mobile APP. Please know that the supplied modification as following is the easiest method not the best method.
```

Comment:
I talked with BLE provision developer of the esp-idf, but they don't think this is an real issue, and they said:
```
The nimble is designed to run users' function in callback, so in some case it allow software costs some time in callback so that lead nimble to be blocked for some time.
```
Beside this, they promise to add non-blocking mode in PC tools and phone APP, in this mode host can continuously to read state of ESP32 and decide how to do next(wait or not) for toit.

Reply:
Why does the C version of the BLE provisioning (the one in the ESP-IDF) not run into this? Is it, because it's not using NimBLE?

Reply:
Because toit sends the BLE message got from NimBLE stack to another task, so the NimBLE stack will not blocked. This cause the host tools will not block by BLE read action, but it must read empty and cause issue.

Reply:
The current implementation does indeed not have the possibility to do "dynamic" reads, where a callback is made to toit to do calculated value that should be the result. We only support "static" reads, where the value to be read is stored in the BLE stack.
@floitsch If there are other places in the platform where there is a callback from the primitive layer to toit, that could be used, but I am not aware that this has been made. Otherwise it would be something like this:
Have a toit function that is called "await_read_request" and blocks on a resource state until a request is received.
Then the driver will need to set the state "read_request_received" and then it will block on a semaphore (with a timeout). When the toit function is done, it will have to call a "provide_read_response" that then releases the semaphore and the primitive will send the response.